File: mgyoutube-multitech/api-node/src/repos/simplisticsearchesdatarepoimpl.ts
import { SearchesDataRepo } from "./searchesdatarepo";


export class SimplisticSearchesDataRepoImpl implements SearchesDataRepo {

    private parentSearches: Map<string, Set<string>> = new Map()

    constructor() {
    }

    async repositoryStartup(): Promise<void> {
        // nothing to do here
    }

    async getSearchesForParentUser(parentUserId: string): Promise<string[]> {
        const returnArray: string[] = [];
        const existingSearches = this.parentSearches.get(parentUserId);

        if (existingSearches != null) {
            existingSearches.forEach(s => returnArray.push(s));
        }

        return returnArray;
    }

    async addSearchToParentUser(parentUserId: string, searchPhrase: string): Promise<void> {
        var existingSet = this.parentSearches.get(parentUserId)
        if (existingSet == null) {
            existingSet = new Set();
            this.parentSearches.set(parentUserId, existingSet);
        }

        existingSet.add(searchPhrase);
    }

    async removeSearchFromParentUser(parentUserId: string, searchPhrase: string): Promise<void> {
        const existingSet = this.parentSearches.get(parentUserId)
        if (existingSet != null) {
            existingSet.delete(searchPhrase);
        }
    }
}

File: mgyoutube-multitech/api-java/src/main/java/com/djpedesen/mgyoutube/api_java/repos/SimplisticSearchesDataRepoImpl.java
package com.djpedesen.mgyoutube.api_java.repos;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.HashSet;
import java.util.List;
import java.util.Map;
import java.util.Set;

public class SimplisticSearchesDataRepoImpl implements SearchesDataRepo {

	private final Map<String, Set<String>> parentSearches = new HashMap<>();

	@Override
	public void repositoryStartup() {
		// nothing to do
	}

	@Override
	public List<String> getSearchesForParentUser(final String parentUserId) {
		if (parentSearches.containsKey(parentUserId)) {
			return new ArrayList<>(parentSearches.get(parentUserId));
		}
		return new ArrayList<>();
	}

	@Override
	public void addSearchToParentUser(final String parentUserId, final String searchPhrase) {
		if (!parentSearches.containsKey(parentUserId)) {
			parentSearches.put(parentUserId, new HashSet<>());
		}

		final Set<String> phrases = parentSearches.get(parentUserId);
		phrases.add(searchPhrase);
	}

	@Override
	public void removeSearchFromParentUser(final String parentUserId, final String searchPhrase) throws Exception {
		if (parentSearches.containsKey(parentUserId)) {
			final Set<String> phrases = parentSearches.get(parentUserId);
			phrases.remove(searchPhrase);
		}
	}
}
